### PR TITLE
[ACTIVITI-3746] - fix form completion with tab fields

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACTIVITI-3746

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/form/components/widgets/core/form-field.model.ts [hunk 1]
```diff
@@ -134,6 +135,20 @@ export class FormFieldModel extends FormWidgetModel {
         return this._isValid;
     }
 
+    isTabVisible() {
```
Comment (eromano) [inline lib/core/form/components/widgets/core/form-field.model.ts]: The visibility doesn't look like a concern of this model what di you think @DenysVuika @mauriziovitale ?

Comment (DenysVuika) [reply lib/core/form/components/widgets/core/form-field.model.ts]: Misses the return value typing. Besides that, the form model already parses tabs and tab fields on init, so I think we are already bound with the notion of tabs. I would suggest leaving here and maybe thinking over splitting in the future

File: lib/core/form/components/widgets/core/form-field.model.ts [hunk 2]
```diff
@@ -209,6 +224,19 @@ export class FormFieldModel extends FormWidgetModel {
         this.updateForm();
     }
 
+    private findFieldByCurrentId(value: any) {
```
Comment (DenysVuika) [inline lib/core/form/components/widgets/core/form-field.model.ts]: misses return type

File: lib/core/form/components/widgets/core/form-field.model.ts (hunk 1 above)
Comment (eromano) [reply lib/core/form/components/widgets/core/form-field.model.ts]: but not visibility

Comment (eromano) [reply lib/core/form/components/widgets/core/form-field.model.ts]: @DenysVuika should it be concern of visibility service?

Comment (DenysVuika) [reply lib/core/form/components/widgets/core/form-field.model.ts]: @eromano probably yes. If that api moves to the visibility service, it should also take `findFieldByCurrentId` function there too

Comment (mauriziovitale) [reply lib/core/form/components/widgets/core/form-field.model.ts]: Yep, I agree it should not be there.

Comment (mauriziovitale) [reply lib/core/form/components/widgets/core/form-field.model.ts]: I see that we already do some checks on tab visibility 
```
  public refreshVisibility(form: FormModel) {
        if (form && form.tabs && form.tabs.length > 0) {
            form.tabs.map((tabModel) => this.refreshEntityVisibility(tabModel));
        }
```

Comment (SilviuCPopa) [reply lib/core/form/components/widgets/core/form-field.model.ts]: The issue it's related to form completion. 
Right now we cannot complete a task that has an invisible tab with an invalid form inside.
If we have just an invisible invalid field we can complete it.
In this PR I added a check to make the field invisible if the tab that belongs to is also invisible.

**Pull request merged**